base/abstractarraymath.jl

```diff
@@ -34,6 +34,7 @@ julia> vec(a)
 """
 vec(a::AbstractArray) = reshape(a,_length(a))
 vec(a::AbstractVector) = a
+vec(a1::AbstractArray, a2::AbstractArray, arest::AbstractArray...) = vcat(vec(a1), vec(a2), map(vec, arest)...)
```

Comment:
I had the same thought... but I thought `vec` should return a ReshapedView, so I don't think it would be too different from hand-rolled loops:

```julia
julia> Base.vec(a1::AbstractArray, a2::AbstractArray, arest::AbstractArray...) = vcat(vec(a1), vec(a2), map(vec, arest)...)
WARNING: Method definition vec(AbstractArray, AbstractArray, AbstractArray...) in module Main at REPL[2]:1 overwritten at REPL[8]:1.

julia> x = rand(10); y=rand(10,10); z=rand(2);

julia> @time vec(x,y,z);
  0.010772 seconds (506 allocations: 26.950 KB)

julia> @time vec(x,y,z);
  0.000011 seconds (10 allocations: 1.281 KB)

julia> x = rand(1000); y=rand(100,100); z=rand(200);

julia> @time vec(x,y,z);
  0.000031 seconds (11 allocations: 87.875 KB)
```

I'm not sure why there's an extra allocation in the second one though...